src/pymedphys/dcm/_level1/dcmdose.py

```diff
@@ -41,12 +40,32 @@
 from ...libutils import get_imports
 IMPORTS = get_imports(globals())
 
-Coords = namedtuple('Coords', 'x y z')
-Dose = namedtuple(
-    'Dose', 'values units type summation heterogeneity_correction')
-
 # pylint: disable=C0103
 
+class DicomDose:
+
+    def __init__(self, dcm_filepath):
+        dcm = pydicom.dcmread(dcm_filepath)
+        if dcm.Modality != "RTDOSE":
+            raise ValueError("The input DICOM file is not an RT Dose file")
+        self.dcm = dcm
+
+        self.values = dcm.pixel_array * dcm.DoseGridScaling
+        self.units = dcm.DoseUnits
+        self.x, self.y, self.z = extract_dicom_patient_xyz(dcm)
+        self.coords = convert_xyz_to_dicom_coords((self.x, self.y, self.z))
+        self.mask = None
+
+    def save_to_dicom(self, filepath):
+        self.dcm.save_as(filepath)
```

Comment:
Maybe just call this function `save_as`? And, I think there is value in thinking of this object as a pydicom object with dose specific extras. So maybe we should expose the pydicom class methods that we envisage useful. `save_as` is certainly one.

What do you think about creating a base dicom wrapper class that exposes the pydicom methods that we want, including having utilities like from_file etc. It would also include some of our other helper methods like from_dict and to_dict. Then we create DicomDose, DicomStructure etc that inherits from our base PyDicomWrapper object. PyDicomWrpper should not inherit from anything within pydicom, just be initialised with a pydicom object and then forward through the pydicom methods.

Let me know what you think :). It's seeming quite nice in my head.

Reply:
Will do! :)